java/ql/test/library-tests/xml/test.xml

```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document>
+    <element attribute="value">Text</element> <!-- $ hasXmlResult -->
```

Comment:
Since this is testing the inline expectation test framework itself, then we should have two more lines: one with a result that doesn't have an expectation comment, and one without a result, but with an expectation comment, such that we get the spurious and missing cases reported in the expected file. Otherwise an empty expected file could just as well mean that the xml isn't seen at all.

Reply:
Good point! Added in ec03ebb, thanks.